R/downloader.R

```diff
@@ -0,0 +1,95 @@
+#' @title Downloader: Downloading National-Scale Hydro feature archives
+#' @description This function downloads and decompresses staged seamless data representing various hydro features in CONUS.
+#' Available options include the seamless NHDPlusV21, the Watershed Boundary dataset and the EPA Reach File (RF). The WBD and RF can be downloaded with no system dependencies.
+#' To download and extract the NHDPlus users need to meet the following requirements: p7zip (MacOS), 7zip (windows)
+#' @param outdir The folder path where data should be downloaded and extracted
+#' @param type The dataset to download and extract. Options include (1) nhdplus (2) wbd (3) rf1
+#' @return the path to the respective geodatabase or e00 file
+#' @export
+#' @importFrom httr progress GET write_disk
+#' @importFrom utils unzip
+#' @importFrom R.utils gunzip
+#' @examples
+#' \dontrun{
+#' outdir = "./data"
+#'   downloader(outdir, "nhdplus")
+#'   downloader(outdir, "wbd")
+#'   downloader(outdir, "rf1")
+#' }
+
+downloader = function(outdir, type = 'rf1') {
+
+  valid_types = c("nhdplus", "wbd", 'rf1')
```

Comment:
Just a note on style... I've used `<-` for assignment everywhere in nhdplusTools.